master/pkg/tasks/task_spec.go

```diff
@@ -131,413 +138,18 @@ func (t *TaskSpec) EnvVars() map[string]string {
-	return env
-}
-
-// EnvVars implements InnerSpec.
-func (s StartTrial) EnvVars(t TaskSpec) map[string]string {
-	portOffset := trialUniquePortOffset(t.Devices)
-	portStr := rendezvousPort(portOffset)
-	return map[string]string{
-		"DET_EXPERIMENT_ID":            fmt.Sprintf("%d", s.InitialWorkload.ExperimentID),
-		"DET_TRIAL_ID":                 fmt.Sprintf("%d", s.InitialWorkload.TrialID),
-		"DET_TRIAL_SEED":               fmt.Sprintf("%d", s.TrialSeed),
-		"DET_EXPERIMENT_CONFIG":        jsonify(s.ExperimentConfig),
-		"DET_HPARAMS":                  jsonify(s.HParams),
-		"DET_INITIAL_WORKLOAD":         jsonify(s.InitialWorkload),
-		"DET_LATEST_CHECKPOINT":        "/run/determined/train/checkpoint.json",
-		"DET_WORKLOAD_MANAGER_TYPE":    string(s.WorkloadManagerType),
-		"DET_RENDEZVOUS_PORT":          strconv.Itoa(portStr),
-		"DET_TRIAL_UNIQUE_PORT_OFFSET": strconv.Itoa(portOffset),
-	}
-}
-
-// LoggingFields implements InnerSpec.
-func (s StartTrial) LoggingFields() map[string]string {
-	return map[string]string{
-		"trial_id": strconv.Itoa(s.InitialWorkload.TrialID),
-	}
-}
-
-// Mounts implements InnerSpec.
-func (s StartTrial) Mounts() []mount.Mount {
-	mounts := ToDockerMounts(s.ExperimentConfig.BindMounts())
-	addMount := func(source, target string, bindOpts *mount.BindOptions) {
-		mounts = append(mounts, mount.Mount{
-			Type: mount.TypeBind, Source: source, Target: target, BindOptions: bindOpts,
-		})
-	}
-
-	if c := s.ExperimentConfig.CheckpointStorage().RawSharedFSConfig; c != nil {
-		addMount(
-			c.HostPath(),
-			model.DefaultSharedFSContainerPath,
-			&mount.BindOptions{Propagation: model.DefaultSharedFSPropagation},
-		)
-	}
-
-	if c := s.ExperimentConfig.DataLayer().RawSharedFSConfig; c != nil {
-		if c.HostStoragePath() != nil && c.ContainerStoragePath() != nil {
-			addMount(*c.HostStoragePath(), *c.ContainerStoragePath(), nil)
-		}
-	}
-	if c := s.ExperimentConfig.DataLayer().RawS3Config; c != nil {
-		if c.LocalCacheHostPath() != nil && c.LocalCacheContainerPath() != nil {
-			addMount(*c.LocalCacheHostPath(), *c.LocalCacheContainerPath(), nil)
-		}
-	}
-	if c := s.ExperimentConfig.DataLayer().RawGCSConfig; c != nil {
-		if c.LocalCacheHostPath() != nil && c.LocalCacheContainerPath() != nil {
-			addMount(*c.LocalCacheHostPath(), *c.LocalCacheContainerPath(), nil)
-		}
-	}
-
-	return mounts
-}
-
-// UseFluentLogging implements InnerSpec.
-func (s StartTrial) UseFluentLogging() bool { return true }
-
-// UseHostMode implements InnerSpec.
-func (s StartTrial) UseHostMode() bool { return s.IsMultiAgent }
-
-// ShmSize implements InnerSpec.
-func (s StartTrial) ShmSize() int64 {
-	if shm := s.ExperimentConfig.Resources().ShmSize(); shm != nil {
-		return int64(*shm)
-	}
-	return 0
-}
-
-// ResourcesConfig implements InnerSpec.
-func (s StartTrial) ResourcesConfig() expconf.ResourcesConfig {
-	return s.ExperimentConfig.Resources()
-}
-
 // TaskSpecMaker is used to make task specs.
 type TaskSpecMaker struct {
```

Comment:
Why is this not just in `core.go`?  It seems like the only `TaskSpecMaker` object is attached to the `master` object.

Reply:
Everything can be moved to `core.go` if you don't split code into packages and everything would be only used in `core.go`. :

Reply:
But literally the only thing that uses this seems to be the master object...